web/src/message_events.ts

```diff
@@ -653,11 +706,14 @@ export function update_messages(events) {
                     )
                 ) {
                     refreshed_current_narrow = true;
-                    message_view.show(current_filter.terms(), {
-                        then_select_id: current_selected_id,
+                    const opts: ShowMessageViewOpts = {
                         trigger: "stream/topic change",
                         force_rerender: true,
-                    });
+                    };
+                    if (current_selected_id !== undefined) {
+                        opts.then_select_id = current_selected_id;
+                    }
+                    message_view.show(current_filter.terms(), opts);
```

Comment:
Or change `then_select_id?: number` to `then_select_id?: number | undefined` in `type ShowMessageViewOpts`.

Reply:
Did this